engine/src/main/java/org/terasology/logic/players/FirstPersonHeldItemMountPointComponent.java

```diff
@@ -15,18 +15,66 @@
  */
 package org.terasology.logic.players;
 
+import jopenvr.VRControllerState_t;
+import org.joml.Matrix4f;
 import org.terasology.entitySystem.Component;
 import org.terasology.entitySystem.Owns;
 import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.math.TeraMath;
 import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.rendering.openvrprovider.ControllerListener;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.math.geom.Quat4f;
 
 /**
  * Only used by the client side so that held items can be positioned in line with the camera
  */
-public class FirstPersonHeldItemMountPointComponent implements Component {
+public class FirstPersonHeldItemMountPointComponent implements Component, ControllerListener {
```

Comment:
Tagging @Josharias and @skaldarnar to review changes to this class as they are much more knowledgeable than I am on ES stuff, i.e. components.